pubtools/_quay/command_executor.py

```diff
@@ -245,3 +258,163 @@ def _run_cmd(self, cmd, err_msg=None, tolerate_err=False, stdin=None):
+            stdin (None|str):
+                This parameter exists only for compatibility with parent class. Sending input
+                to containers is not supported.
+
+        Returns (str, str):
+            Tuple of stdout and stderr generated by the command.
+        """
+        err_msg = err_msg or "An error has occured when executing a command."
+        cmd_exec = self.client.exec_create(self.container["Id"], cmd)
+        stdout, stderr = self.client.exec_start(cmd_exec["Id"], demux=True)
+
+        if self.client.exec_inspect(cmd_exec["Id"]).get("ExitCode") != 0 and not tolerate_err:
+            LOG.error("Command {0} failed with {1}".format(cmd, stderr))
+            raise RuntimeError(err_msg)
+
+        if stdout is None:
+            stdout = b""
+        if stderr is None:
+            stderr = b""
+        out_str = stdout.decode("utf-8")
+        err_str = stderr.decode("utf-8")
+
+        return (out_str, err_str)
+
+    def _add_file(self, data, file_name):
+        """
+        Add a text file to the running container.
+
+        The primary use-case is to store a secret which will be accessed from inside the container.
+        File will be stored in the path /tmp/<file_name>.
+
+        Args:
+            data (str):
+                Data that should be stored in the container.
+            file_name (str):
+                Name of the file.
+        """
+        data_stream = io.BytesIO()
+        data_tar = tarfile.TarFile(fileobj=data_stream, mode="w")
+        encoded_data = data.encode("utf-8")
+        tarinfo = tarfile.TarInfo(name=file_name)
+        tarinfo.size = len(encoded_data)
+        tarinfo.mtime = time.time()
+        data_tar.addfile(tarinfo, io.BytesIO(encoded_data))
+        data_tar.close()
+
+        data_stream.seek(0)
+        success = self.client.put_archive(
+            container=self.container["Id"], path="/tmp", data=data_stream
+        )
+
+        if not success:
+            raise RuntimeError("File was not successfully added to the container")
+
+    def skopeo_login(self, username=None, password=None):
+        """
+        Attempt to login to Quay if no login credentials are present.
+
+        This method is reimplemented because it uses a different approach to input the password.
+
+        Args:
+            username (str):
+                Username for login.
+            password (str):
+                Password for login.
+        """
+        cmd_check = "skopeo login --get-login quay.io"
+        out, err = self._run_cmd(cmd_check, tolerate_err=True)
+        if username and username in out:
+            LOG.info("Already logged in to Quay.io")
+            return
+
+        if not username or not password:
+            raise ValueError(
+                "Skopeo login credentials are not present. Quay user and token must be provided."
+            )
+        LOG.info("Logging in to Quay with provided credentials")
+
+        password_file = "skopeo_password.txt"
```

Comment:
did you consider using docker ENV ?

Reply:
I suppose it would be a viable solution as well, though I would prefer to keep the current one, simply because it's already implemented and tested. Is that ok?

Reply:
sure, I was just wondering if was any problem with ENV or anything which prevented you from using it